If we have a thumbnail for a saved link, display it

rnagle opened the pull request:
This is meant to address #25.

The pull request includes:

- Updated default link HTML to include a place marker (`#!IMAGE!#`) for the featured image.
- Updated the tag documentation on the options page to indicate what `#!IMAGE!#` is used for.
- Lightweight styling to make sure links display in some sensical way when a featured image is available.

See:

![image](https://cloud.githubusercontent.com/assets/72608/9476039/18480f7e-4b30-11e5-900f-52e7cd4e1f21.png)

aschweigert commented:
This looks fine to me. So this addresses the appearance in roundup posts and on the saved links archive page, but does this also show the thumbnail in the saved links widget (when one is available)? We might want to add a widget option (similar to other widgets in largo) to not show the thumbnail there if a particular site doesn't want them, but other than that this seems good to go.

rnagle commented:
Added some stuff to the saved links widget to include featured images when available. Also added the ability to show/hide featured images via the widget settings.

I kept the CSS for both the widget, the archive page and the roundup post content as lean as possible to avoid a situation where styling the plugin components via theme stylesheet requires super specific selectors or lots of `!important` declarations.
